jac-uk/admin
# #257 Refine database security rules and write more tests

## What's included?
This work fixes the email verification issue with third party (specifically Microsoft) authenticated accounts to the admin platform. The security rules require for all users that an email address is verified, and there is no way of telling on firestore where the request originated from.

Microsoft accounts do not set the email verified flag to true. It is a requirements that these are set to true if logging in via the admin platform, as users from the JAC and other government departments logging into the platform should not have to verify their email address.

This pull request is one half of the work to fix this issue, by checking the authentication provider and if the email verified flag is false, requesting a cloud function (with proper access to update all google authentication service user details) to verify their email.

The other half of this is https://github.com/jac-uk/digital-platform/pull/633.

This should also fix the case where an existing user has logged in - it'll verify their email when the SPA is loaded.

## Who should test?

- [x] Product Owner
- [ ]  UTG

## How to test?

Login using a microsoft account. Once the digital platform work is merged as well, it should work. Unfortunately the only way to tell if it doesn't work as if you login using a MS account, and it gives you permission denied.

## Risk - how likely is this to impact other areas?

🟠 **Medium risk - this does change code that is shared with other areas**

## Additional context
Include screen grabs, video demo, notes etc.

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
